src/ring_theory/ideal/basic.lean

```diff
@@ -303,6 +305,56 @@ end pi
 
 end ideal
 
+/-- A right ideal in a semiring `R` is an additive submonoid `s` such that
+`b * a ∈ s` whenever `b ∈ s`. If `R` is a ring, then `s` is an additive subgroup.  -/
+@[reducible] def right_ideal (R : Type u) [semiring R] := submodule Rᵐᵒᵖ R
+
+namespace right_ideal
+
+variables [semiring α] (I : right_ideal α) {a b : α}
+
+protected lemma zero_mem : (0 : α) ∈ I := I.zero_mem
+
+protected lemma add_mem : a ∈ I → b ∈ I → a + b ∈ I := I.add_mem
+
+variables (a)
+lemma mul_mem_right : b ∈ I → b * a ∈ I := I.smul_mem (mul_opposite.op a)
+variables {a}
+
+@[ext] lemma ext {I J : right_ideal α} (h : ∀ x, x ∈ I ↔ x ∈ J) : I = J :=
+submodule.ext h
+
+lemma sum_mem {ι : Type*} {t : finset ι} {f : ι → α} :
+  (∀c∈t, f c ∈ I) → (∑ i in t, f i) ∈ I := submodule.sum_mem I
+
+theorem eq_top_of_unit_mem
+  (x y : α) (hx : x ∈ I) (h : x * y = 1) : I = ⊤ :=
+eq_top_iff.2 $ λ z _, calc
+    z = (x * y) * z : by simp [h]
+  ... = x * (y * z) : mul_assoc x y z
+  ... ∈ I : I.mul_mem_right _ hx
+
+theorem eq_top_of_is_unit_mem {x} (hx : x ∈ I) (h : is_unit x) : I = ⊤ :=
+let ⟨y, hy⟩ := h.exists_right_inv in eq_top_of_unit_mem I x y hx hy
+
+theorem eq_top_iff_one : I = ⊤ ↔ (1:α) ∈ I :=
+⟨by rintro rfl; trivial,
+ λ h, eq_top_of_unit_mem _ _ 1 h (by simp)⟩
+
+theorem ne_top_iff_one : I ≠ ⊤ ↔ (1:α) ∉ I :=
+not_congr I.eq_top_iff_one
+
+/-- A right ideal is a left ideal of the opposite (semi)ring. -/
+@[simps] def to_ideal : ideal αᵐᵒᵖ :=
```

Comment:
I can see this being more useful as an equiv or even order_iso between `ideal αᵐᵒᵖ` and `right_ideal α`. In particular, having things like `to_ideal.injective` and `to_ideal.map_bot` seem valuable.

Reply:
In the same vein, `ideal.to_right_ideal : ideal α \equivo right_ideal αᵐᵒᵖ` would go in the other direction